spec/index.bs

```diff
@@ -1326,6 +1331,77 @@ dictionary GPUBufferBinding {
 
   * {{GPUBufferBinding/size}}: If undefined, specifies the range starting at {{GPUBufferBinding/offset}} and ending at the end of the buffer.
 
+A {{GPUBindGroup}} object has the following internal slots:
+
+<dl dfn-type=attribute dfn-for="GPUBindGroup">
+    : <dfn>\[[layout]]</dfn> of type {{GPUBindGroupLayout}}>.
+    ::
+        The {{GPUBindGroupLayout}} this {{GPUBindGroup}} associates.
+
+    : <dfn>\[[bindings]]</dfn> of type sequence<{{GPUBindGroupBinding}}>.
+    ::
+        The set of {{GPUBindGroupBinding}}s this {{GPUBindGroup}} describes.
+</dl>
+
+### {{GPUDevice/createBindGroup()|GPUDevice.createBindGroup(GPUBindGroupDescriptor)}} ### {#GPUDevice-createBindGroup}
+
+<div algorithm="GPUDevice.createBindGroup(descriptor)">
+
+The <dfn method for="GPUDevice">createBindGroup(|descriptor|)</dfn> method is used to create {{GPUBindGroup}}s.
+
+1. Ensure [=bind group device validation=] is not violated.
+1. Let |bindGroup| be a new valid {{GPUBindGroup}} object.
+1. Ensure {{GPUBindGroupLayout}} in |descriptor|.{{GPUBindGroupDescriptor/layout}} is valid.
+1. Ensure the number of {{GPUBindGroupLayoutDescriptor/bindings}} of |descriptor|.{{GPUBindGroupDescriptor/layout}}
+    exactly equals to the number of |descriptor|.{{GPUBindGroupDescriptor/bindings}}.
+1. For each {{GPUBindGroupBinding}} |bindingDescriptor| in |descriptor|.{{GPUBindGroupDescriptor/bindings}}:
+    1. Ensure |bindingDescriptor|.{{GPUBindGroupBinding/binding}} does not violate [=bind group binding validation=].
+    1. If |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is {{GPUSampler}}:
+        1. Ensure it is a valid {{GPUSampler}} object.
+        1. Ensure [=sampler binding validation=] is not violated.
+    1. If |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is {{GPUTextureView}}:
+        1. Ensure it is a valid {{GPUTextureView}} object.
+        1. Ensure [=texture view binding validation=] is not violated.
+    1. If |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is {{GPUBufferBinding}}
+        1. Ensure it is a valid {{GPUBufferBinding}} object.
+        1. Ensure [=buffer binding validation=] is not violated.
+    1. Insert |bindingDescriptor| into |bindGroup|.{{GPUBindGroup/[[bindings]]}}.
+1. Return |bindGroup|.
```

Comment:
`bindGroup.[[layout]]` is not set.

I would move the creation of `bindGroup` down here. I don't think it even needs multiple steps, e.g.:

```
1. Return a new {{GPUBindGroup}} object with:
    - {{GPUBindGroup/[[layout]]}} = |descriptor|.{{GPUBindGroupDescriptor/layout}}
    - {{GPUBindGroup/[[bindings]]}} = |descriptor|.{{GPUBindGroupDescriptor/bindings}}
```

Reply:
Done. Good point.